packages/@lwc/style-compiler/src/transform.ts

```diff
@@ -58,5 +64,11 @@ export function transform(src: string, id: string, config: Config = {}): { code:
 
     const result = postcss(plugins).process(src, { from: id }).sync();
 
-    return { code: serialize(result, config) };
+    const code = serialize(result, config);
+
+    if (isAPIFeatureEnabled(APIFeature.VALIDATE_CSS_CONTENT_IN_STYLE_COMPILER, apiVersion)) {
+        validateStyleTextContents(code, parseFragment);
+    }
```

Comment:
@nolanlawson  do you think it would be worth creating a separate flag to let the compiler know when it's compiling for SSR?

We could avoid calling the validation for non-SSR environments and we could avoid creating the static content tokens in the browser too.

Reply:
Right now we compile for both targets. So the compiler has to serve both unfortunately.

In the future we definitely want to have the possibility of multiple targets, but that will be part of the new SSR compiler effort.